pkg/policycache/cache.go

```diff
@@ -33,11 +34,11 @@ type policyCache struct {
 type Interface interface {
 	Add(policy *kyverno.ClusterPolicy)
 	Remove(policy *kyverno.ClusterPolicy)
-	Get(pkey PolicyType, nspace *string) []*kyverno.ClusterPolicy
+	Get(pkey PolicyType, kind *string, nspace *string) ([]string, []*kyverno.ClusterPolicy)
```

Comment:
A few questions:
1. The first return value "names" is not used anywhere, do we really need it?
2. Why do take `kind` and `nspace` as references?
3. Can we add a comment on how cpol and pol can be fetched using this method? Something like  "if nspace set, it returns pol (though here the type is ClusterPolicy)"